### PR TITLE
Roll Flutter Engine from c50eb8a65097 to 419fb8c0ab3e

### PR DESCRIPTION
**Pull request by a-siva**
https://github.com/flutter/engine/compare/c50eb8a65097...419fb8c0ab3e

2024-09-06 98614782+auto-submit[bot]@users.noreply.github.com Reverts "[engine] always force platform channel responses to schedule a task. (https://github.com/flutter/flutter/issues/54975)" (https://github.com/flutter/engine/pull/55000)
2024-09-06 [skia-flutter-autoroll@skia.org](mailto:skia-flutter-autoroll@skia.org) Roll Skia from b6bab0fde426 to 6ad117bd2efe (2 revisions) (https://github.com/flutter/engine/pull/54999)
2024-09-06 [skia-flutter-autoroll@skia.org](mailto:skia-flutter-autoroll@skia.org) Roll Fuchsia Test Scripts from D9INMR2u4wcyiZ750... to 5dqcFlKzRjJb6V95W... (https://github.com/flutter/engine/pull/54998)
2024-09-06 [skia-flutter-autoroll@skia.org](mailto:skia-flutter-autoroll@skia.org) Roll Skia from a09312b70d37 to b6bab0fde426 (3 revisions) (https://github.com/flutter/engine/pull/54997)
2024-09-06 [skia-flutter-autoroll@skia.org](mailto:skia-flutter-autoroll@skia.org) Roll Skia from 368f209ccca5 to a09312b70d37 (1 revision) (https://github.com/flutter/engine/pull/54995)
2024-09-06 [skia-flutter-autoroll@skia.org](mailto:skia-flutter-autoroll@skia.org) Roll Skia from aec11ae18bb6 to 368f209ccca5 (3 revisions) (https://github.com/flutter/engine/pull/54992)
2024-09-06 [skia-flutter-autoroll@skia.org](mailto:skia-flutter-autoroll@skia.org) Roll Fuchsia Linux SDK from xNv47d1TZmK9XgTxu... to PBeI0gGvgFdXV6hCg... (https://github.com/flutter/engine/pull/54990)
2024-09-06 [skia-flutter-autoroll@skia.org](mailto:skia-flutter-autoroll@skia.org) Roll Skia from 809f868ded1c to aec11ae18bb6 (22 revisions) (https://github.com/flutter/engine/pull/54988)
2024-09-06 [30870216+gaaclarke@users.noreply.github.com](mailto:30870216+gaaclarke@users.noreply.github.com) Removes the int storage from Color (https://github.com/flutter/engine/pull/54714)
2024-09-06 [chris@bracken.jp](mailto:chris@bracken.jp) iOS,macOS: Add logging of duplicate codesign binaries (https://github.com/flutter/engine/pull/54987)
2024-09-06 [skia-flutter-autoroll@skia.org](mailto:skia-flutter-autoroll@skia.org) Roll Fuchsia Test Scripts from k4lKsecg0pdIp-U7c... to D9INMR2u4wcyiZ750... (https://github.com/flutter/engine/pull/54984)
2024-09-05 [a-siva@users.noreply.github.com](mailto:a-siva@users.noreply.github.com) Manual roll of Dart. (https://github.com/flutter/engine/pull/54983)
2024-09-05 [chris@bracken.jp](mailto:chris@bracken.jp) iOS,macOS: add unsigned_binaries.txt (https://github.com/flutter/engine/pull/54977)
2024-09-05 [jason-simmons@users.noreply.github.com](mailto:jason-simmons@users.noreply.github.com) Manual Skia roll to 809f868ded1c (https://github.com/flutter/engine/pull/54972)
2024-09-05 [1961493+harryterkelsen@users.noreply.github.com](mailto:1961493+harryterkelsen@users.noreply.github.com) [canvaskit] Fix incorrect calculation of ImageFilter paint bounds (https://github.com/flutter/engine/pull/54980)
2024-09-05 [jonahwilliams@google.com](mailto:jonahwilliams@google.com) [engine] always force platform channel responses to schedule a task. (https://github.com/flutter/engine/pull/54975)
2024-09-05 [tugorez@users.noreply.github.com](mailto:tugorez@users.noreply.github.com) Fix unexpected ViewFocus events when Text Editing utilities change focus in the middle of a blur call. (https://github.com/flutter/engine/pull/54965)

Also rolling transitive DEPS:
fuchsia/sdk/core/linux-amd64 from xNv47d1TZmK9 to PBeI0gGvgFdX

### CODE REVIEW

**Comment by christopherfujino** [comment]
I'm working on fixing this PR

File: packages/flutter_tools/lib/src/update_packages_pins.dart [hunk 1]
```diff
@@ -28,6 +28,7 @@ const Map<String, String> kManuallyPinnedDependencies = <String, String>{
   'leak_tracker': '10.0.7', // https://github.com/flutter/devtools/issues/3951
   'leak_tracker_testing': '3.0.1', // https://github.com/flutter/devtools/issues/3951
   'leak_tracker_flutter_testing': '3.0.8', // https://github.com/flutter/devtools/issues/3951
+  'path_provider_android': '2.2.1', // https://github.com/flutter/flutter/issues/140796
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/update_packages_pins.dart:31]: FYI @goderbauer & @gmackall I re-pinned path_provider_android to unblock this roll :'(

**Comment by christopherfujino** [comment]
A little bit of context, package:_macros recently bumped its version and package:analyzer was forced to raise its lower constraint on it: https://github.com/dart-lang/sdk/commit/5285faad677344009ae3c043b4643c4986a48284#diff-a8403f1fc69a35a5ee70b920d57c1d402b06e5d10929ec8daedfebbf8c34c42fR16

When this Dart SDK change was attempted in an engine -> framework autoroll, it failed pub solving in the flutter_tools: https://github.com/flutter/flutter/pull/154731 because flutter_tools did not yet have the newer version of analyzer.

I resolved the issue by manually running `flutter update-packages --force-upgrade` on a branch with the engine revision that had the newer version of `_macros`.

**Comment by christopherfujino** [comment]
https://ci.chromium.org/ui/p/flutter/builders/try/Windows%20framework_tests_libraries/68219/infra

```
The following TestFailure was thrown running a test:
Expected: Color:<Color(0xff19e600)>
  Actual: Color:<Color(0xff1ae600)>

When the exception was thrown, this was the stack:
#4      main.<anonymous closure> (file:///C:/b/s/w/ir/x/w/flutter/packages/flutter/test/painting/decoration_image_lerp_test.dart:183:7)
<asynchronous suspension>
#5      testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:189:15)
<asynchronous suspension>
#6      TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:1032:5)
<asynchronous suspension>
<asynchronous suspension>
(elided one frame from package:stack_trace)

This was caught by the test expectation on the following line:
  file:///C:/b/s/w/ir/x/w/flutter/packages/flutter/test/painting/decoration_image_lerp_test.dart line 183
The test description was:
  ImageDecoration.lerp 1
```

**Comment by zanderso** [comment]
> https://ci.chromium.org/ui/p/flutter/builders/try/Windows%20framework_tests_libraries/68219/infra
> 
> ```
> The following TestFailure was thrown running a test:
> Expected: Color:<Color(0xff19e600)>
>   Actual: Color:<Color(0xff1ae600)>
> 
> When the exception was thrown, this was the stack:
> #4      main.<anonymous closure> (file:///C:/b/s/w/ir/x/w/flutter/packages/flutter/test/painting/decoration_image_lerp_test.dart:183:7)
> <asynchronous suspension>
> #5      testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:189:15)
> <asynchronous suspension>
> #6      TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:1032:5)
> <asynchronous suspension>
> <asynchronous suspension>
> (elided one frame from package:stack_trace)
> 
> This was caught by the test expectation on the following line:
>   file:///C:/b/s/w/ir/x/w/flutter/packages/flutter/test/painting/decoration_image_lerp_test.dart line 183
> The test description was:
>   ImageDecoration.lerp 1
> ```

@gaaclarke Wondering if this could be due to https://github.com/flutter/engine/pull/54714 ?

**Comment by a-siva** [comment]
> https://ci.chromium.org/ui/p/flutter/builders/try/Windows%20framework_tests_libraries/68219/infra
> 
> ```
> The following TestFailure was thrown running a test:
> Expected: Color:<Color(0xff19e600)>
>   Actual: Color:<Color(0xff1ae600)>
> 
> When the exception was thrown, this was the stack:
> #4      main.<anonymous closure> (file:///C:/b/s/w/ir/x/w/flutter/packages/flutter/test/painting/decoration_image_lerp_test.dart:183:7)
> <asynchronous suspension>
> #5      testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:189:15)
> <asynchronous suspension>
> #6      TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:1032:5)
> <asynchronous suspension>
> <asynchronous suspension>
> (elided one frame from package:stack_trace)
> 
> This was caught by the test expectation on the following line:
>   file:///C:/b/s/w/ir/x/w/flutter/packages/flutter/test/painting/decoration_image_lerp_test.dart line 183
> The test description was:
>   ImageDecoration.lerp 1
> ```

this could be the golden file issue that @gaaclarke was talking about

**Comment by gaaclarke** [comment]
> this could be the golden file issue that @gaaclarke was talking about

It's not quite what I was expecting.  I had to fix one of these types of failures to get the PR through presubmit checks.  Here's how I addressed it: https://github.com/flutter/flutter/pull/154680/files

I'm not sure why the test runs here, but not in presubmit.  I would be fine just updating that assertion or making it fuzzy like I did above as part of this roll.

**Comment by a-siva** [comment]
> > this could be the golden file issue that @gaaclarke was talking about
> 
> It's not quite what I was expecting. I had to fix one of these types of failures to get the PR through presubmit checks. Here's how I addressed it: https://github.com/flutter/flutter/pull/154680/files
> 
> I'm not sure why the test runs here, but not in presubmit. I would be fine just updating that assertion or making it fuzzy like I did above as part of this roll.

@gaaclarke can you make your suggested change in this PR so we can rerun the tests and see if that works out.

**Comment by zanderso** [comment]
Oops. I pushed a change that just naively updates the test expectation, but looking at the previous addition of a new matcher, I suspect that's not quite right.

**Comment by gaaclarke** [comment]
I don't have permissions to push to this branch.  Here's the patch:
```
commit 9095aa51144561d0d094c08f1a718bf30f6c1bc3
Author: Aaron Clarke <aaclarke@google.com>
Date:   Fri Sep 6 11:41:54 2024 -0700

    added fuzzy color matching

diff --git a/packages/flutter/test/painting/decoration_image_lerp_test.dart b/packages/flutter/test/painting/decoration_image_lerp_test.dart
index ab24e4fdc4..ad9fe6e064 100644
--- a/packages/flutter/test/painting/decoration_image_lerp_test.dart
+++ b/packages/flutter/test/painting/decoration_image_lerp_test.dart
@@ -15,6 +15,36 @@ import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_test/flutter_test.dart';
 
+// TODO(gaaclarke): Unify the different instances of _ColorMatcher.
+/// Positive result if the colors would be mapped to the same argb8888 color.
+class _ColorMatcher extends Matcher {
+  _ColorMatcher(this._target);
+
+  final ui.Color _target;
+
+  @override
+  Description describe(Description description) {
+    return description.add('matches "$_target"');
+  }
+
+  @override
+  bool matches(dynamic item, Map<dynamic, dynamic> matchState) {
+    if (item is ui.Color) {
+      return item.colorSpace == _target.colorSpace &&
+          (item.a - _target.a).abs() <= 0.004 &&
+          (item.r - _target.r).abs() <= 0.004 &&
+          (item.g - _target.g).abs() <= 0.004 &&
+          (item.b - _target.b).abs() <= 0.004;
+    } else {
+      return false;
+    }
+  }
+}
+
+Matcher _matchesColor(ui.Color color) {
+  return _ColorMatcher(color);
+}
+
 void main() {
   testWidgets('ImageDecoration.lerp 1', (WidgetTester tester) async {
     final MemoryImage green = MemoryImage(Uint8List.fromList(<int>[
@@ -178,17 +208,17 @@ void main() {
         return getPixel(x, y);
       }
       const Color lime = Color(0xFF00FF00);
-      expect(getBlockPixel(0), lime); // pure green
-      expect(getBlockPixel(1), lime); // 100% green 0% red
-      expect(getBlockPixel(2), const Color(0xFF1AE600));
-      expect(getBlockPixel(3), const Color(0xFF33CC00));
-      expect(getBlockPixel(4), const Color(0xFF808000)); // 50-50 mix green/red
-      expect(getBlockPixel(5), const Color(0xFFCD3200));
-      expect(getBlockPixel(6), const Color(0xFFE61900));
-      expect(getBlockPixel(7), const Color(0xFFFF0000)); // 0% green 100% red
-      expect(getBlockPixel(8), const Color(0xFFFF0000)); // pure red
+      expect(getBlockPixel(0), _matchesColor(lime)); // pure green
+      expect(getBlockPixel(1), _matchesColor(lime)); // 100% green 0% red
+      expect(getBlockPixel(2), _matchesColor(const Color(0xFF1AE600)));
+      expect(getBlockPixel(3), _matchesColor(const Color(0xFF33CC00)));
+      expect(getBlockPixel(4), _matchesColor(const Color(0xFF808000))); // 50-50 mix green/red
+      expect(getBlockPixel(5), _matchesColor(const Color(0xFFCD3200)));
+      expect(getBlockPixel(6), _matchesColor(const Color(0xFFE61900)));
+      expect(getBlockPixel(7), _matchesColor(const Color(0xFFFF0000))); // 0% green 100% red
+      expect(getBlockPixel(8), _matchesColor(const Color(0xFFFF0000))); // pure red
       for (int index = 9; index < 40; index += 1) {
-        expect(getBlockPixel(index), lime);
+        expect(getBlockPixel(index), _matchesColor(lime));
       }
     }
```

**Comment by gaaclarke** [comment]
@zanderso how come you can push?  This is what I got:
```
$ git push a-siva
error: Authentication error: Authentication required: You must have push access to verify locks
error: failed to push some refs to 'github.com:a-siva/flutter.git'
```

**Comment by zanderso** [comment]
1. Click on the link for the branch on @a-siva's fork above `a-siva:tot`.
2. Navigate to the file to edit, and edit with the GitHub editor.
3. Click the button to push to the branch.

**Comment by a-siva** [comment]
> I don't have permissions to push to this branch. Here's the patch:
> 
> ```
> commit 9095aa51144561d0d094c08f1a718bf30f6c1bc3
> Author: Aaron Clarke <aaclarke@google.com>
> Date:   Fri Sep 6 11:41:54 2024 -0700
> 
>     added fuzzy color matching
> 
> diff --git a/packages/flutter/test/painting/decoration_image_lerp_test.dart b/packages/flutter/test/painting/decoration_image_lerp_test.dart
> index ab24e4fdc4..ad9fe6e064 100644
> --- a/packages/flutter/test/painting/decoration_image_lerp_test.dart
> +++ b/packages/flutter/test/painting/decoration_image_lerp_test.dart
> @@ -15,6 +15,36 @@ import 'package:flutter/foundation.dart';
>  import 'package:flutter/material.dart';
>  import 'package:flutter_test/flutter_test.dart';
>  
> +// TODO(gaaclarke): Unify the different instances of _ColorMatcher.
> +/// Positive result if the colors would be mapped to the same argb8888 color.
> +class _ColorMatcher extends Matcher {
> +  _ColorMatcher(this._target);
> +
> +  final ui.Color _target;
> +
> +  @override
> +  Description describe(Description description) {
> +    return description.add('matches "$_target"');
> +  }
> +
> +  @override
> +  bool matches(dynamic item, Map<dynamic, dynamic> matchState) {
> +    if (item is ui.Color) {
> +      return item.colorSpace == _target.colorSpace &&
> +          (item.a - _target.a).abs() <= 0.004 &&
> +          (item.r - _target.r).abs() <= 0.004 &&
> +          (item.g - _target.g).abs() <= 0.004 &&
> +          (item.b - _target.b).abs() <= 0.004;
> +    } else {
> +      return false;
> +    }
> +  }
> +}
> +
> +Matcher _matchesColor(ui.Color color) {
> +  return _ColorMatcher(color);
> +}
> +
>  void main() {
>    testWidgets('ImageDecoration.lerp 1', (WidgetTester tester) async {
>      final MemoryImage green = MemoryImage(Uint8List.fromList(<int>[
> @@ -178,17 +208,17 @@ void main() {
>          return getPixel(x, y);
>        }
>        const Color lime = Color(0xFF00FF00);
> -      expect(getBlockPixel(0), lime); // pure green
> -      expect(getBlockPixel(1), lime); // 100% green 0% red
> -      expect(getBlockPixel(2), const Color(0xFF1AE600));
> -      expect(getBlockPixel(3), const Color(0xFF33CC00));
> -      expect(getBlockPixel(4), const Color(0xFF808000)); // 50-50 mix green/red
> -      expect(getBlockPixel(5), const Color(0xFFCD3200));
> -      expect(getBlockPixel(6), const Color(0xFFE61900));
> -      expect(getBlockPixel(7), const Color(0xFFFF0000)); // 0% green 100% red
> -      expect(getBlockPixel(8), const Color(0xFFFF0000)); // pure red
> +      expect(getBlockPixel(0), _matchesColor(lime)); // pure green
> +      expect(getBlockPixel(1), _matchesColor(lime)); // 100% green 0% red
> +      expect(getBlockPixel(2), _matchesColor(const Color(0xFF1AE600)));
> +      expect(getBlockPixel(3), _matchesColor(const Color(0xFF33CC00)));
> +      expect(getBlockPixel(4), _matchesColor(const Color(0xFF808000))); // 50-50 mix green/red
> +      expect(getBlockPixel(5), _matchesColor(const Color(0xFFCD3200)));
> +      expect(getBlockPixel(6), _matchesColor(const Color(0xFFE61900)));
> +      expect(getBlockPixel(7), _matchesColor(const Color(0xFFFF0000))); // 0% green 100% red
> +      expect(getBlockPixel(8), _matchesColor(const Color(0xFFFF0000))); // pure red
>        for (int index = 9; index < 40; index += 1) {
> -        expect(getBlockPixel(index), lime);
> +        expect(getBlockPixel(index), _matchesColor(lime));
>        }
>      }
> ```

I will apply your patch and psu it, not sure why you are getting the authentication error.

File: packages/flutter/test/painting/decoration_image_lerp_test.dart [hunk 2]
```diff
@@ -178,17 +208,17 @@ void main() {
...
-      expect(getBlockPixel(2), const Color(0xFF19E600));
-      expect(getBlockPixel(3), const Color(0xFF33CC00));
-      expect(getBlockPixel(4), const Color(0xFF808000)); // 50-50 mix green/red
-      expect(getBlockPixel(5), const Color(0xFFCD3200));
-      expect(getBlockPixel(6), const Color(0xFFE61900));
-      expect(getBlockPixel(7), const Color(0xFFFF0000)); // 0% green 100% red
-      expect(getBlockPixel(8), const Color(0xFFFF0000)); // pure red
+      expect(getBlockPixel(0), _matchesColor(lime)); // pure green
+      expect(getBlockPixel(1), _matchesColor(lime)); // 100% green 0% red
+      expect(getBlockPixel(2), _matchesColor(const Color(0xFF1AE600)));
```
Comment (gaaclarke) [inline packages/flutter/test/painting/decoration_image_lerp_test.dart:213]: nit: It's not a huge deal, but reverting this to the original color is a bit cleaner.

**Review by gaaclarke — APPROVED** [review]

File: packages/flutter/test/painting/decoration_image_lerp_test.dart (hunk 2 above)
Comment (a-siva) [reply packages/flutter/test/painting/decoration_image_lerp_test.dart]: Done.

**Comment by jtmcdole** [comment]
>   Expected: contains '(((((RELOAD WORKED)))))'

**Comment by zanderso** [comment]
> > Expected: contains '(((((RELOAD WORKED)))))'

That text appears to be in the logs, but it might be racing with the check due to coming too late.

**Comment by a-siva** [comment]
> > > Expected: contains '(((((RELOAD WORKED)))))'
> 
> That text appears to be in the logs, but it might be racing with the check due to coming too late.

I am going to re-run that check to see if it passes.

**Comment by a-siva** [comment]
@christopherfujino do you think we should revert the line that updates video_player_android to 2.7.3 (switch back to 2.7.2), the Windows build seems to get this error:

[        ] Execution failed for task ':video_player_android:compileDebugJavaWithJavac'.
[        ] > Could not resolve all files for configuration ':video_player_android:debugCompileClasspath'.
[        ]    > Could not resolve androidx.media3:media3-exoplayer:1.4.1.
[        ]      Required by:
[        ]          project :video_player_android
[        ]       > Cannot find a version of 'androidx.media3:media3-exoplayer' that satisfies the version constraints:
[        ]            Dependency path 'io.flutter.plugins.videoplayer:video_player_android:1.0-SNAPSHOT' --> 'androidx.media3:media3-exoplayer:1.4.1'
[        ]            Constraint path 'io.flutter.plugins.videoplayer:video_player_android:1.0-SNAPSHOT' --> 'androidx.media3:media3-exoplayer:{strictly 1.4.0}' because of the following reason: dependency was locked to version '1.4.0'

**Comment by christopherfujino** [comment]
> @christopherfujino do you think we should revert the line that updates video_player_android to 2.7.3 (switch back to 2.7.2), the Windows build seems to get this error:
> 
> [ ] Execution failed for task ':video_player_android:compileDebugJavaWithJavac'. [ ] > Could not resolve all files for configuration ':video_player_android:debugCompileClasspath'. [ ] > Could not resolve androidx.media3:media3-exoplayer:1.4.1. [ ] Required by: [ ] project :video_player_android [ ] > Cannot find a version of 'androidx.media3:media3-exoplayer' that satisfies the version constraints: [ ] Dependency path 'io.flutter.plugins.videoplayer:video_player_android:1.0-SNAPSHOT' --> 'androidx.media3:media3-exoplayer:1.4.1' [ ] Constraint path 'io.flutter.plugins.videoplayer:video_player_android:1.0-SNAPSHOT' --> 'androidx.media3:media3-exoplayer:{strictly 1.4.0}' because of the following reason: dependency was locked to version '1.4.0'

When I try to run the generate_gradle_lockfiles script on this branch, I get the following error:

```
Processing /usr/local/google/home/fujino/git/flutter/dev/benchmarks/test_apps/stocks/android
Unhandled exception:
ProcessException:
FAILURE: Build failed with an exception.

* What went wrong:
Could not open cp_settings generic class cache for settings file '/usr/local/google/home/fujino/git/flutter/dev/benchmarks/test_apps/stocks/android/settings.gradle' (/usr/local/google/home/fujino/.gradle/caches/8.3/scripts/639158s6nc50atry7sibozeyt).
> BUG! exception in phase 'semantic analysis' in source unit '_BuildScript_' Unsupported class file major version 66
```

@gmackall help! Is this an issue with my local setup, or with our deps on this branch?

Here's my android toolchain I have locally:

```
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /usr/local/google/home/fujino/Android/Sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: /usr/local/google/home/fujino/Downloads/android-studio/jbr/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.9+0-17.0.9b1087.7-11185874)
    • All Android licenses accepted.
```

**Comment by christopherfujino** [comment]
Ok, Gray helped unblock me, I'm re-generating gradle lockfiles presently. Will push an update when it's done.

**Review by matanlurey — APPROVED** [review]

**Pull request merged**